bundles/org.openhab.binding.publictransportswitzerland/src/main/java/org/openhab/binding/publictransportswitzerland/internal/stationboard/PublicTransportSwitzerlandStationboardHandler.java

```diff
@@ -0,0 +1,187 @@
+            "stationboard/number",
+            "stationboard/stop/departureTimestamp",
+            "stationboard/stop/delay",
+            "stationboard/stop/platform");
+
+    private final ChannelGroupUID dynamicChannelGroupUID = new ChannelGroupUID(getThing().getUID(), "departures");
+
+    private final Logger logger = LoggerFactory.getLogger(PublicTransportSwitzerlandStationboardHandler.class);
+
+    private @Nullable ScheduledFuture<?> updateDataJob;
+
+    public PublicTransportSwitzerlandStationboardHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // This handler does not support any commands
+    }
+
+    @Override
+    public void initialize() {
+        PublicTransportSwitzerlandStationboardConfiguration config = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+
+        if (config.station == null || config.station.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            updateDataJob = scheduler.scheduleWithFixedDelay(this::updateData, 0, 60, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (updateDataJob != null && !updateDataJob.isCancelled()) {
+            updateDataJob.cancel(true);
+        }
+    }
+
+    public void updateData() {
+        PublicTransportSwitzerlandStationboardConfiguration config = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+
+        try {
+            String escapedStation = URLEncoder.encode(config.station, StandardCharsets.UTF_8.name());
+            String requestUrl = BASE_URL + "stationboard?station=" + escapedStation + fieldFilters;
+
+            String response = HttpUtil.executeUrl("GET", requestUrl, 10_000);
+            logger.debug("Got response from API: {}", response);
+
+            JsonElement jsonObject = new JsonParser().parse(response);
+
+            updateChannels(jsonObject);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            logger.warn("Unable to fetch stationboard data", e);
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            updateState(CHANNEL_TSV, new StringType("No data available"));
+        }
+    }
+
+    private static String createFilterForFields(String... fields) {
+        return Arrays.stream(fields).map((field) -> "&fields[]=" + field).collect(Collectors.joining());
+    }
+
+    private void updateChannels(JsonElement jsonObject) throws Exception {
+        JsonArray stationboard = jsonObject.getAsJsonObject().get("stationboard").getAsJsonArray();
+        createDynamicChannels(stationboard.size());
+
+        List<String> tsvRows = new ArrayList<>();
+
+        for (int i = 0; i < stationboard.size(); i++) {
```

Comment:
Take care, if the answer returns less departures than your current number of channels, you have to set the last channels to UNDEF.

Reply:
10279b2